components/RuiExpandCollapse/src/RuiExpandCollapse.ts

```diff
@@ -0,0 +1,209 @@
+      this._collapseableEl.classList.remove('hide-content');
+  
+      const transitionEndHandler = (): void => {
+        if (this._collapseableEl) {
+          this._collapseableEl.style.height = '';
+          
+          this._collapseableEl.removeEventListener('transitionend', transitionEndHandler);
+        }
+      }
+  
+      this._collapseableEl.addEventListener('transitionend', transitionEndHandler);
+    }
+  }
+
+  /**
+   * Initialises the expand collapse logic and styling,
+   * once complete makes the expand collapse visible
+   */
+  private _initialiseExpandCollapse(): void {
+    if (this.shadowRoot) {
+      this._collapseableEl = this.shadowRoot.querySelector('.details-container');
+
+      if (this._collapseableEl) {
+        
+        // need to set height initially if closed without triggering animation
+        if (!this.open) {
+          this._collapseableEl.style.height = '0px';
+          this._collapseableEl.hidden = true;
+          this._collapseableEl.classList.add('hide-content');
+        }
+        
+        let expandCollapse: HTMLElement | null = this.shadowRoot.querySelector('.expand-collapse');
+        if (expandCollapse) {
+          expandCollapse.style.opacity = '1';
+        }
+      }
+    }
+  }
+  
+  /**
+   * After initial render initialise expand collapse logic. Because
+   * we are animating slotted content, we have to wait for the 
+   * slot to be mounted
+   */
+  public firstUpdated(): void {
+    if (this.shadowRoot) {
+      this._detailsSlotEl = this.shadowRoot.querySelector('#details-slot');
+
+      if (this._detailsSlotEl) {
+        // when the slotted content changes we initialise expand collapse
+        // we need to wait for this because the animation of heigh calc 
+        // will only work once the slot and it's content have mounted and rendered
+        this._detailsSlotEl.addEventListener('slotchange', (): void => {
+          this._initialiseExpandCollapse();
+         });
+      }
+    }
+  }
+
+  public updated(changedProperties: Map<string, RuiExpandCollapsePropertyType>): void {
+    changedProperties.forEach((oldValue: RuiExpandCollapsePropertyType, propName: string): void => {
+      // detect change in open prop and trigger animation as necessary
+      if (propName === 'open' && this._collapseableEl) {
+        // transition from closed to open
+        if (this.open && !oldValue) {
+          this._triggerExpandAnimation();
+        }
+         
+        // transition from open to closed
+        if (!this.open && oldValue) {
+          this._triggerCollapseAnimation();
+        }
+      }
+    });
+  }
+
+  /**
+   * Render method
+   */  
+  public render(): TemplateResult {
+    return html`
+      <section class=${`expand-collapse${this.open ? ' is-open' : ''}`}>
+        <button id="expand-trigger" @click="${this._handleClick}" class="summary-container" aria-expanded=${`${this.open ? 'true': 'false'}`} aria-controls="expandable-section">
```

Comment:
Almost there! The button ID and the matching `aria-labelledby` values need to be a unique set per instance so avoid multiples of the same IDs.

Reply:
Also now we're using`button`, you'll need to remove the default browser styles (namely border)

Reply:
@ttaiyo Okay I've added a UUID generator to keep the id's unique per instance and have removed the border default styling. I've also created a variable for the default browser outline when focusing so users can override/disable as needed